controlplane/kubeadm/internal/cluster.go

```diff
@@ -98,6 +99,12 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 		return nil, &RemoteClusterConnectionError{Name: clusterKey.String(), Err: err}
 	}
 
+	// Create a metadata-only client.
+	metadataClient, err := metadata.NewForConfig(restConfig)
+	if err != nil {
+		return nil, &RemoteClusterConnectionError{Name: clusterKey.String(), Err: err}
+	}
```

Comment:
We shouldn't need to pass the `restConfig`, or the client-go plain metadata client. 

Instead, we can do something like the following:

```
out := metav1.PartialObjectMetadata{}
					out.SetGroupVersionKind(schema.GroupVersionKind{
						Group:   "apiextensions...",
						Version: "v1",
						Kind:    "CustomResourceDefinition",
					})
				key := client.ObjectKey{Namespace: ns, Name: dep.Name}
				err = w.Client.Get(ctx, key, &out)
```

Reply:
One other thing we need to make sure of is to never cache CRDs, which we can do when we create the client for the workload cluster by adding the CRD object to `util.DelegatingClientFuncWithUncached(`